### DISCUSSION TITLE
Clock oscillator config bits

### DISCUSSION THREAD

**Discussion by ldicdican** [CommunityGD32Cores/GD32Core-New#88] [General]
GD32F450 has internal oscillators as follows:

Internal 8 MHz factory-trimmed RC and external 4 to 32 MHz crystal oscillator
 Internal 48 MHz RC oscillator
 Internal 28 MHz RC oscillator
 Internal 40 KHz RC calibrated oscillator and external 32.768 KHz crystal oscillators

How can we set on Arduino if I will use external crystal oscillator?

**Comment by maxgerhardt** [comment]
Arduino on GD32450 is not yet implemented here o_o

https://github.com/CommunityGD32Cores/ArduinoCore-GD32/tree/main/system

If you meant F350, the current default already **is** HXTAL

https://github.com/CommunityGD32Cores/ArduinoCore-GD32/blob/c2bfff7740ec2dc58478a5eddaa537c49b7df885/system/GD32F3x0_firmware/CMSIS/GD/GD32F3x0/Source/system_gd32f3x0.c#L56-L68

Dynamic change of clock settings via menu selection or build flags is still an open issue per https://github.com/CommunityGD32Cores/ArduinoCore-GD32/issues/19

**Reply by ldicdican** [reply]
Yeah I mean F350, sorry for typo and thanks for directing me on the config file.
